app/src/test/java/com/keylesspalace/tusky/SpanUtilsTest.kt

```diff
@@ -1,182 +1,187 @@
-        val inputSpannable = FakeSpannable(input)
-        highlightSpans(inputSpannable, 0xffffff)
-        val spans = inputSpannable.spans
-        Assert.assertEquals(6, spans.size)
+/** The [Pattern.UNICODE_CHARACTER_CLASS] flag is not supported on Android, on Android it is just always on.
+ * Since thesse tests run on a regular Jvm, we need a to set this flag or they would behave differently.
+ * */
+private val urlPattern = Regex.VALID_URL_PATTERN_STRING.toPattern(Pattern.CASE_INSENSITIVE or Pattern.UNICODE_CHARACTER_CLASS)
+private val tagPattern = TAG_PATTERN_STRING.toPattern(Pattern.CASE_INSENSITIVE or Pattern.UNICODE_CHARACTER_CLASS)
+private val mentionPattern = MENTION_PATTERN_STRING.toPattern(Pattern.CASE_INSENSITIVE or Pattern.UNICODE_CHARACTER_CLASS)
+
+val finders = listOf(
+    PatternFinder("http", FoundMatchType.HTTPS_URL, urlPattern),
+    PatternFinder("#", FoundMatchType.TAG, tagPattern),
+    PatternFinder("@", FoundMatchType.MENTION, mentionPattern)
+)
+
+@RunWith(Parameterized::class)
+class SpanUtilsTest(
+    private val stringToHighlight: String,
+    private val highlights: List<Pair<Int, Int>>
+) {
+
+    companion object {
+        @Parameterized.Parameters(name = "{0}")
+        @JvmStatic
+        fun data() = listOf(
+            arrayOf("@mention", listOf(0 to 8)),
+            arrayOf("@mention@server.com", listOf(0 to 19)),
+            arrayOf("#tag", listOf(0 to 4)),
+            arrayOf("#tåg", listOf(0 to 4)),
+            arrayOf("https://thr.ee/meh?foo=bar&wat=@at#hmm", listOf(0 to 38)),
+            arrayOf("http://thr.ee/meh?foo=bar&wat=@at#hmm", listOf(0 to 37)),
+            arrayOf(
+                "one #one two: @two three : https://thr.ee/meh?foo=bar&wat=@at#hmm four #four five @five 6 #six",
+                listOf(4 to 8, 14 to 18, 27 to 65, 71 to 76, 82 to 87, 90 to 94)
+            ),
+            arrayOf("http://first.link https://second.link", listOf(0 to 17, 18 to 37)),
+            arrayOf("#test", listOf(0 to 5)),
+            arrayOf(" #AfterSpace", listOf(1 to 12)),
+            arrayOf("#BeforeSpace ", listOf(0 to 12)),
+            arrayOf("@#after_at", listOf(1 to 10)),
+            arrayOf("あいうえお#after_hiragana", listOf<Pair<Int, Int>>()),
+            arrayOf("##DoubleHash", listOf(1 to 12)),
+            arrayOf("###TripleHash", listOf(2 to 13)),
+            arrayOf("something#notAHashtag", listOf<Pair<Int, Int>>()),
+            arrayOf("test##maybeAHashtag", listOf(5 to 19)),
+            arrayOf("testhttp://not.a.url.com", listOf<Pair<Int, Int>>()),
+            arrayOf("test@notAMention", listOf<Pair<Int, Int>>()),
+            arrayOf("test@notAMention#notAHashtag", listOf<Pair<Int, Int>>()),
+            arrayOf("test@notAMention@server.com", listOf<Pair<Int, Int>>()),
+            // Mastodon will not highlight this mention, although it would be valid according to their regex
+            // arrayOf("@test@notAMention@server.com", listOf<Pair<Int, Int>>()),
```

Comment:
This is the only failing test now, the old code failed at least 18 testcases (I added more after trying it)